node/lib/util/submodule_change.js

```diff
@@ -45,20 +45,31 @@ class SubmoduleChange {
      * null `oldSha` implies that the submodule was added, a null `newSha`
      * implies that it was removed, and if neither is null, the submodule was
      * changed.
+     * 
+     * UPDATE: the change used to contain only their sha, because default our
+     * sha can be obtained from head commit. An explicit field is added for
+     * cases when our sha is not necessarily the same as the head.
      *
      * @param {String | null} oldSha
      * @param {String | null} newSha
+     * @param {String | null | undefined} ourSha
```

Comment:
Are we really distinguishing between null and undefined?  That seems bad.

Reply:
I guess you're trying to allow old callers of this function to not be modified.  Instead, please just change them.

Reply:
yah, I was trying to avoid making changes in 40 places

Reply:
What about:
```
     * @param {String | null} oldSha sha from which changes are computed
     * @param {String | null} newSha sha to which changes are computed
     * @param {String | null} ourSha sha agains which changes will be applied
```

Reply:
nit: s/agains/against/

But I think the text in the most-recent version is pretty close to OK.  I would change it to:
"In a 3 way merge, `oldSha` is the merge base, `newSha`  is the right side of the merge and `ourSha`  is left side."    

Also, are these shas?  Or could they be refs?  If they are just shas, could they be https://www.nodegit.org/api/oid/ instead of string?